app/controllers/VersionSwitch.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import play.api.mvc.{Action, Request}
+
+import scala.concurrent.Future
+import scala.xml.NodeSeq
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+trait VersionSwitch {
+ self: BackendController =>
+
+  def versionSwitch(v1Action: Action[NodeSeq], v2Action: Action[NodeSeq]): Action[NodeSeq] =
+    Action.async(parse.xml) {
+    (request: Request[NodeSeq]) =>
```

Comment:
type superfluous.

Reply:
It's not necessarily. When originally building this, we were having type inference issues causing the compiler to pick the wrong overload to `Action#apply`, as a `Request[_]` is also a `RequestHeader`. I haven't checked it here, but I'd imagine that it's possible the same thing is happening here.

This is a case where I think I'd prefer to keep it to ensure the compiler does the right thing -- even if removing it now works for some reason.

Reply:
Not sure about that - if the inferencer can't determine the type, it will let us know!
But ok - leave it.

Reply:
In this case, it wasn't that it couldn't infer a type, it's that it infered a super type of what it actually was - we needed to correct that.